### DISCUSSION TITLE
IPython Extensions Not Used

### DISCUSSION THREAD

**Discussion by jcarpinelli-bdai** [quarto-dev/quarto-cli#9825] [Q&A]
### Bug description

When writing a ` .qmd` file, IPython extensions appear not to be used. For example, I am trying to use the `nb_mypy` extension to type-check all code blocks. Executing the code blocks in a notebook (`.ipynb` file) shows that the extension is running. When I write `%load_ext nb_mypy` in a `.qmd` file though, type checks are not conducted.

### Steps to reproduce

First, `pip install nb_mypy`. Then, `quarto render` the following document. No errors will be shown in the rendered output. If you `quarto convert` the document to a notebook, then open and execute the notebook, you will see errors thrown by `nb_mypy`.

`````
---
title: Type Checks in an Executable Document
jupyter: python3
---

# Interfaces

````{python}
#| echo: false
#| output: false
#| error: false
%load_ext nb_mypy
%nb_mypy On
````

It would be really cool to type-check interfaces which are documented via Quarto! For example, the following code block should cause the document to display runtime errors.

````{python}
def f() -> bool:
    return "true" # an error should be thrown here!
````

`````

### Expected behavior

The document show errors as output values, just as the document shows in an `IPython` notebook.

### Actual behavior

No errors are shown in the rendered output.

### Your environment

- IDE: VSCode
- System: Ubuntu 22.04 (Docker container)

### Quarto check output

```bash
> quarto check 

Quarto 1.4.554
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.554
      Path: /opt/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2024.05
      Chromium: 869685

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /home/dev/.TinyTeX/bin/x86_64-linux
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.12
      Path: /workspaces/asi/venv/bin/python3
      Jupyter: 5.7.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/

```

**Comment by cscheid** [comment]
I don't think we document support for IPython extensions anywhere in our docs. Why do you expect this to work?

**Comment by cderv** [comment]
When I run your example it works I believe on my side 
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/095162d5-ffb4-4731-a922-62dc6b139c6e)

You may need to update the python dependencies. As said, this is not specific quarto support I believe but we do use some tech that may now handle this better (nbconvert maybe ? nbclient maybe ?). 

Anyhow, I believe this is working
